src/common/AndroidUtils.ts

```diff
@@ -154,157 +139,147 @@ export class AndroidSDKUtils {
         }
 
         if (!AndroidSDKUtils.isCached()) {
-            try {
-                const stdout = CommonUtils.executeCommandSync(
-                    `${AndroidSDKUtils.getSdkManagerCommand()} --list`
-                );
-                if (stdout) {
-                    const packages = AndroidPackages.parseRawPackagesString(
-                        stdout
-                    );
-                    AndroidSDKUtils.packageCache = packages;
-                }
-            } catch (err) {
-                return Promise.reject(err);
-            }
+            return CommonUtils.executeCommandAsync(
+                `${AndroidSDKUtils.getSdkManagerCommand()} --list`
+            )
+                .then((result) => {
+                    if (result.stdout && result.stdout.length > 0) {
+                        const packages = AndroidPackages.parseRawPackagesString(
+                            result.stdout
+                        );
+                        AndroidSDKUtils.packageCache = packages;
+                    }
+                    return Promise.resolve(AndroidSDKUtils.packageCache);
+                })
+                .catch((error) => Promise.reject(error));
+        } else {
+            return Promise.resolve(AndroidSDKUtils.packageCache);
         }
-
-        return Promise.resolve(AndroidSDKUtils.packageCache);
     }
 
-    public static fetchEmulators(): Promise<AndroidVirtualDevice[]> {
-        return new Promise((resolve, reject) => {
-            let devices: AndroidVirtualDevice[] = [];
-            try {
-                const result = CommonUtils.executeCommandSync(
-                    AndroidSDKUtils.getAvdManagerCommand() + ' list avd'
-                );
-                if (result) {
+    public static async fetchEmulators(): Promise<AndroidVirtualDevice[]> {
+        let devices: AndroidVirtualDevice[] = [];
+        return CommonUtils.executeCommandAsync(
+            AndroidSDKUtils.getAvdManagerCommand() + ' list avd'
+        )
+            .then((result) => {
+                if (result.stdout && result.stdout.length > 0) {
                     devices = AndroidVirtualDevice.parseRawString(
-                        result.toString()
+                        result.stdout
```

Comment:
Nice to learn that the default encoding of `child_process.exec()` is 'utf8', returning strings instead of `Buffer`s (the default of `execSync`). A lot less cumbersome to deal with in our use case.